### PR TITLE
Add and configure Hangfire

### PR DESCRIPTION
**Pull request by jicata**
https://github.com/SoftUni-Internal/suls-issues/issues/2916

### CODE REVIEW

File: Open Judge System/OJS.Common/BackgroundJobs/HangfireBackgroundJobService.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    using System;
+    using System.Linq.Expressions;
+
+    using global::Hangfire;
+    using OJS.Common.BackgroundJobs.Contracts;
+
+    public class HangfireBackgroundJobService : IBackgroundJobService
+    {
+        public void AddOrUpdateRecurringJob(object recurringJobId, Expression<Action> methodCall, string cronExpression)
+            => RecurringJob.AddOrUpdate((string)recurringJobId, methodCall, cronExpression);
```
Comment (rozay) [inline Open Judge System/OJS.Common/BackgroundJobs/HangfireBackgroundJobService.cs]: `=>` on the upper line, if its more than 120 characters, separate parameters on new lines.

File: Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+{
+    using System.Web.Hosting;
+
+    using Hangfire;
+
+    public class HangfireBootstrapper : IRegisteredObject
+    {
+        private static readonly object LockObject = new object();
+
+        public static HangfireBootstrapper instance;
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs]: Better make this field private.

File: Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   
+        private bool started;
+
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs]: Unnecessary blank line

File: Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        public void Start()
+        {
+            lock (LockObject)
+            {
+                if (!this.started)
+                {
+                    this.started = true;
+
+                    HostingEnvironment.RegisterObject(this);
+
```
Comment (KristianMariyanov) [inline Open Judge System/Web/OJS.Web/App_Start/HangfireConfiguration/HangfireBootstrapper.cs]: remove one line

File: Open Judge System/OJS.Common/BackgroundJobs/HangfireBackgroundJobService.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿namespace OJS.Common.BackgroundJobs
```
Comment (KristianMariyanov) [inline Open Judge System/OJS.Common/BackgroundJobs/HangfireBackgroundJobService.cs]: This is service. Better make separate OJS.Services.Common Project

**Pull request merged**